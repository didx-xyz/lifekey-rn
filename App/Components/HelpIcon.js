// external dependencies
import React from "react"
import Svg, { Circle, Path } from "react-native-svg"
import PropTypes from "prop-types"

const HelpIcon = (props) => {
  const dynamic = {
    "stroke": props.stroke
  }

  return (
    <Svg width={props.width} height={props.height} version="1.1" xmlns="http://www.w3.org/2000/svg" xmlnsXlink="http://www.w3.org/1999/xlink" viewBox="0 0 28.2 29.8" xmlSpace="preserve">
        <Circle {...styles.cls1} {...dynamic} cx="16" cy="16" r="9"/>
        <Path {...styles.cls1} {...dynamic} d="M18.915,13.512
          c0,2.575-2.915,1.838-2.915,5.151"/>
        <Path {...styles.cls1} {...dynamic} d="M13.085,13.512
          c0-1.61,1.305-2.915,2.915-2.915s2.915,1.305,2.915,2.915"/>
        <Circle {...styles.cls1} {...dynamic} cx="16.03" cy="20.874" r="0.528"/>

    </Svg>
  )
}

HelpIcon.defaultProps = {
  "stroke": "#000"
}

HelpIcon.propTypes = {
  "width": PropTypes.number.isRequired,
  "height": PropTypes.number.isRequired,
  "stroke": PropTypes.string
}

const styles = {
  "cls1": {
    "fill": "none",
    "stroke": "#000",
    "strokeMiterlimit": 10
  }
}

export default HelpIcon
