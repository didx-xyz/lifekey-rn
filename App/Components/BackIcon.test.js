// external dependencies
import React from "react"

// internal dependencies
import HelpIcon from "./HelpIcon"

test("HelpIcon renders correctly, by default", () => {
  const tree = shallow(
    <HelpIcon width={16} height={16} />
  )

  expect(tree).toMatchSnapshot()
})

test("HelpIcon renders correctly, when modified", () => {
  const tree = shallow(
    <HelpIcon width={32} height={32} stroke="#fff" />
  )

  expect(tree).toMatchSnapshot()
})
