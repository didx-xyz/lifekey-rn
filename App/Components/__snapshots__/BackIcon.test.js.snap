// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BackIcon renders correctly, by default 1`] = `
<Svg
  height={16}
  preserveAspectRatio="xMidYMid meet"
  viewBox="0 0 9.4 17.7"
  width={16}
  xmlns="http://www.w3.org/2000/svg"
>
  <Polyline
    fill="none"
    points="8.9,0.5 0.5,9 8.7,17.2"
    stroke="#000"
    strokeLinecap="round"
    strokeLinejoin="round"
  />
</Svg>
`;

exports[`BackIcon renders correctly, when modified 1`] = `
<Svg
  height={32}
  preserveAspectRatio="xMidYMid meet"
  viewBox="0 0 9.4 17.7"
  width={32}
  xmlns="http://www.w3.org/2000/svg"
>
  <Polyline
    fill="none"
    points="8.9,0.5 0.5,9 8.7,17.2"
    stroke="#fff"
    strokeLinecap="round"
    strokeLinejoin="round"
  />
</Svg>
`;
